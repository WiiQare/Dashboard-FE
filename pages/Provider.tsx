import React, { useState } from "react";


let sidebarAction: boolean

export default function Providers() {

    const [open, setOpen] = useState(sidebarAction = false)

    const handleSidebarState = (): void => {
        setOpen(!open)
    }
    return (
        <></>
    );
}


