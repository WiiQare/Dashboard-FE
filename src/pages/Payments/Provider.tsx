import { UserContext } from '@/context/UserContext';
import Router from 'next/router';
import React, { useEffect, useLayoutEffect, useState } from 'react';
import { fetchData } from '../api/fetchData';
import CardsData from "@/data/tableData/payments/paymentsCards";
import Pagination from "@/components/atom/pagination";
import Content from '@/components/content';
import PayersColumns from '@/data/tableData/payments/provider/providerColumns';
import Loader from '@/components/atom/loader';
interface UserInterface {
    type: string;
    userId: string;
    phoneNumber: string;
    names: string;
    email: string;
    access_token: string;
}

let newSkip: number = 0;

const Provider = () => {
    const User = React.useContext(UserContext);
    const [userState, setUserState] = useState<UserInterface | null>(User?.user);
    const [userAuth, setUserAuth] = useState<boolean | undefined>(User?.authenticated);
    const [tableData, setTableData] = useState<any>(null);
    const [summary, setSummary] = useState<any>();
    const [numOfItems, setNumOfItems] = useState<number>(0); // Set initial value to 0
    const [cardData, setCardData] = useState<any[]>([]);
    const [currentPage, setCurrentPage] = useState<number>(1);
    const [pageSize, setPageSize] = useState<number>();

    const [mounted, setMounted] = useState<boolean>(false);

    const take = 10;
    const skip = 0;
    useLayoutEffect(() => {
        setUserAuth(Boolean(sessionStorage.getItem("userAuth")));
        setUserState(JSON.parse(sessionStorage.getItem("userState") || 'null'));
        setMounted(true);
    }, [User?.authenticated, User?.user]);

    useEffect(() => {
        if (!mounted) return; // Return early if the component is not mounted

        const fetchDataAsync = async () => {
            const res = await fetchData("/payments/providers", userState?.access_token, take, skip);
            const summaryData = await fetchData("/payments/summary", userState?.access_token);
            setTableData(res);
            setSummary(summaryData);
            // console.log("res", res);
        };
        if (mounted && userAuth) {
            fetchDataAsync();

        }

    }, [mounted, userAuth, userState?.access_token]); // Remove other dependencies to fetch data only once when mounted



    useEffect(() => {
        if (summary) {
            setCardData(CardsData(summary))
            setNumOfItems(summary.numberOfRows | 10)
            // console.log('CardsData:', CardsData(summary));
            // console.log('numOfItems:', numOfItems);
        }
    }, [summary]);

    const handlePageChange = async (page: number) => {
        if (page >= 1 && page <= Math.ceil(numOfItems / take)) {
            setCurrentPage(page);
            const newSkip = (page - 1) * take;
            const res = await fetchData('/payments/providers', userState?.access_token, take, newSkip);
            setTableData(res);
        }
    };
    const handlePageSizeChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
        const newPageSize = parseInt(event.target.value);
        setPageSize(newPageSize);

        fetchData('/payments/providers', userState?.access_token, newPageSize, 0).then((res) => {
            setTableData(res);
            setCurrentPage(1);
        });
    };

    if (!tableData || !summary) {
        return <Loader />
    }

    return (
        <div>
            <Content columns={PayersColumns} data={tableData} cardsData={cardData} groups={[]}>
                <div className="flex">
                    <div>
                        <div className="flex items-center mt-3 mr-2">
                            <div className="mr-4">
                                <label htmlFor="pageSize">Items per Page:</label>
                                <select
                                    id="pageSize"
                                    className="ml-2 p-1 border border-gray-300 rounded"
                                    value={pageSize}
                                    onChange={handlePageSizeChange}
                                >
                                    <option value={10}>10</option>
                                    <option value={20}>20</option>
                                    <option value={30}>30</option>
                                    {/* Add more options as needed */}
                                </select>
                            </div>
                        </div>
                    </div>
                    <Pagination
                        currentPage={currentPage}
                        totalPages={Math.ceil(numOfItems / take)}
                        onPageChange={handlePageChange}
                    />
                </div>
            </Content>
            <h1 className="bg-red-500 w-28"> </h1>
        </div>
    );
};

export default Provider;
