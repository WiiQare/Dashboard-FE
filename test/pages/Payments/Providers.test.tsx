import { render } from "@testing-library/react";
import Provider from "../../../src/pages/Payments/Providers";



describe('Provider Component', () => {
  it('renders the component without errors', async () => {
    render(<Provider />);
  });

});