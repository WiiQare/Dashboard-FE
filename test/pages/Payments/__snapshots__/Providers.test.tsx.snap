// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Vouchers component 1`] = `
<div>
  <div
    class="transition-1 overflow-y-auto p-4 h-full w-full sm:px-4"
    id="page-skeleton"
  >
    <div
      class="grid grid-cols-1 gap-4 mb-4"
    >
      <div
        class="w-full"
      >
        <div
          class="grid cards grid-cols-1 gap-2 md:grid-cols-2 md:gap-6 xl:grid-cols-5 2xl:gap-7.5 p-[23px]"
        >
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3"
              >
                <div
                  class="w-16 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-24 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-36 h-4 bg-gray-300 rounded-md "
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3"
              >
                <div
                  class="w-16 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-24 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-36 h-4 bg-gray-300 rounded-md "
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3"
              >
                <div
                  class="w-16 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-24 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-36 h-4 bg-gray-300 rounded-md "
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3"
              >
                <div
                  class="w-16 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-24 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-36 h-4 bg-gray-300 rounded-md "
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3"
              >
                <div
                  class="w-16 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-24 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-36 h-4 bg-gray-300 rounded-md "
                />
              </div>
            </div>
          </div>
          <div
            class="border-[#180a0a07] border-[0.2px] shadow-md sm:grid-cols-2 items-left rounded-md dark:bg-[#182644] bg-white p-[20px]"
          >
            <div
              class="flex flex-row items-center justify-center h-full space-x-5 animate-pulse dark:bg-[#182644] bg-white"
            >
              <div
                class="flex-shrink-0 w-12 h-12 rounded-full dark:bg-gray-700 bg-gray-300"
              />
              <div
                class="flex flex-col space-y-3"
              >
                <div
                  class="w-16 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-24 h-4 bg-gray-300 rounded-md "
                />
                <div
                  class="w-36 h-4 bg-gray-300 rounded-md "
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="transition-c-0.5"
        >
          <div
            class="rounded-2xl border-none bg-white transition-c-0.5 dark:bg-[#111827] no-scrollbar overflow-x-auto border-gray-300 dark:border-gray-700 shadow-md"
          >
            <div
              class="animate-pulse"
            >
              <div
                class="w-full h-12 bg-gray-300 dark:bg-gray-700"
              />
              <div
                class="overflow-x-auto"
              >
                <table
                  class="min-w-full divide-y divide-gray-300 dark:divide-gray-700"
                >
                  <thead
                    class="p-2"
                  >
                    <tr>
                      <th
                        class=""
                      >
                        <div
                          class="my-1 ml-1 w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </th>
                      <th
                        class=""
                      >
                        <div
                          class="my-1 ml-1 w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </th>
                      <th
                        class=""
                      >
                        <div
                          class="my-1 ml-1 w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </th>
                      <th
                        class=""
                      >
                        <div
                          class="my-1 ml-1 w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </th>
                      <th
                        class=""
                      >
                        <div
                          class="my-1 ml-1 w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </th>
                      <th
                        class=""
                      >
                        <div
                          class="my-1 ml-1 w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="bg-white dark:dark:bg-[#111827] divide-y divide-gray-300 dark:divide-gray-700"
                  >
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                    <tr>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                      <td
                        class="py-4 px-4"
                      >
                        <div
                          class="w-3/4 h-4 bg-gray-300 dark:bg-gray-700"
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  )
</div>
`;
